### PR TITLE
fix bug in event source mapping with custom IAM role

### PR DESCRIPTION
**Pull request by PetrKohut**
## What did you implement:
I put a condition whether we have custom IAM role or not in order to create a proper EventSourceMapping template.

Fixes [#2813](https://github.com/serverless/serverless/issues/2813)

## How did you implement it:
I simply check provider.role and function.role whether we have custom IAM role in ARN format `arn:aws:iam::account:role/foo` and if so, I change DependOn to be empty array, instead of `IamPolicyLambdaExecution`. Empty array behaves like there wouldn't be DependOn at all.
Note: I decided to go this way to keep `streamTemplate` the same without condition concatenating of strings. 

## How can we verify it:
We should be able to successfully build this serverless.yaml script:
```yaml
service: hello-service

provider:
  name: aws
  runtime: nodejs4.3
  cfLogs: true
  role: arn:aws:iam::account:role/foo

functions:
  hello:
    handler: handler.hello
    events:
      - stream:
          arn: arn:aws:kinesis:region:account:stream/bar
          batchSize: 100
          startingPosition: LATEST
```

## Todos:

- [x] Write tests
- [ ] Write documentation
- [ ] Fix linting errors
- [x] Make sure code coverage hasn't dropped
- [ ] Provide verification config/commands/resources
- [x] Enable ["Allow edits from maintainers"](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) for this PR
- [x] Change ready for review message below

***Is this ready for review?:*** YES

### CODE REVIEW

File: lib/plugins/aws/deploy/compile/events/stream/index.js [hunk 1]
```diff
@@ -65,10 +65,16 @@ class AwsCompileStreamEvents {
             const streamLogicalId = this.provider.naming
               .getStreamLogicalId(functionName, streamType, streamName);
 
+            const funcRole = this.serverless.service.provider.role || functionObj.role;
+            const dependsOn = (typeof funcRole !== 'object'
```
Comment (pmuens) [inline lib/plugins/aws/deploy/compile/events/stream/index.js]: Could you rewrite this in an easier to read way with `if` and `else`?

Comment (PetrKohut) [inline lib/plugins/aws/deploy/compile/events/stream/index.js]: Fixed.

**Comment by PetrKohut** [comment]
I've added tests as well.

**Comment by pmuens** [comment]
Thanks @PetrKohut that looks really good! 💯 

Will review it tomorrow morning in depth!

**Comment by pmuens** [comment]
Thank you for the updates @PetrKohut 

GTM from my side! Great job 👍

**Pull request merged**